test/cppsuite/tests/search_near_01.cxx

```diff
@@ -0,0 +1,257 @@
+        /* Check the prefix length is not greater than the key size. */
+        testutil_assert(key_size >= PREFIX_KEY_LEN);
+
+        logger::log_msg(LOG_INFO,
+          "Populate configuration with key size: " + std::to_string(key_size) +
+            " key count: " + std::to_string(key_count) +
+            " number of collections: " + std::to_string(collection_count));
+
+        /* Create n collections as per the configuration. */
+        for (uint64_t i = 0; i < collection_count; ++i)
+            /*
+             * The database model will call into the API and create the collection, with its own
+             * session.
+             */
+            database.add_collection();
+
+        /* Spawn 26 threads to populate the database. */
+        for (uint64_t i = 0; i < ALPHABET.size(); ++i) {
+            thread_context *tc = new thread_context(i, thread_type::INSERT, config,
+              connection_manager::instance().create_session(), tsm, tracking, database);
+            workers.push_back(tc);
+            tm.add_thread(populate_worker, tc, ALPHABET, PREFIX_KEY_LEN);
+        }
+
+        /* Wait for our populate threads to finish and then join them. */
+        logger::log_msg(LOG_INFO, "Populate: waiting for threads to complete.");
+        tm.join();
+
+        /* Cleanup our workers. */
+        for (auto &it : workers) {
+            delete it;
+            it = nullptr;
+        }
+
+        /* Force evict all the populated keys in all of the collections. */
+        int cmpp;
+        scoped_session session = connection_manager::instance().create_session();
+        for (uint64_t count = 0; count < collection_count; ++count) {
+            collection &coll = database.get_collection(count);
+            scoped_cursor evict_cursor =
+              session.open_scoped_cursor(coll.name.c_str(), "debug=(release_evict=true)");
+
+            for (uint64_t i = 0; i < ALPHABET.size(); ++i) {
+                for (uint64_t j = 0; j < ALPHABET.size(); ++j) {
+                    for (uint64_t k = 0; k < ALPHABET.size(); ++k) {
+                        std::string key = {ALPHABET.at(i), ALPHABET.at(j), ALPHABET.at(k)};
+                        evict_cursor->set_key(evict_cursor.get(), key.c_str());
+                        evict_cursor->search_near(evict_cursor.get(), &cmpp);
+                        testutil_check(evict_cursor->reset(evict_cursor.get()));
+                    }
+                }
+            }
+        }
+        srchkey_len = random_generator::instance().generate_integer(1UL, PREFIX_KEY_LEN);
+        logger::log_msg(LOG_INFO, "Populate: finished.");
+    }
+
+    void
+    read_operation(test_harness::thread_context *tc) override final
+    {
+
+        logger::log_msg(
+          LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+
+        std::map<uint64_t, scoped_cursor> cursors;
+        tc->stat_cursor = tc->session.open_scoped_cursor(STATISTICS_URI);
```

Comment:
Is there any chance that the thread context already has stat cursor?

Reply:
To allow the usage of runtime_monitor::get_stat() to be thread safe, tc->stat_cursor was added to the thread context. Having a discussion with @luke-pearson , opening a cursor for every thread wouldn't be lightweight. Therefore I thought that a thread will need to perform cursor open manually for the stat file.

Reply:
OK, do we need to check `if (tc->stat_cursor == NULL)` before opening another one?

Reply:
Nice catch! Don't think this should be happening at all, since read_operation() is the thread caller. I'm happy to place an assert to ensure this.